src/traces/scattergl/convert.js

```diff
@@ -111,7 +113,27 @@ function convertTextStyle(trace) {
     var optsOut = {};
     var i;
 
-    optsOut.text = trace.text;
+    var texttemplate = trace.texttemplate;
+    if(texttemplate) {
+        var d3locale = gd._fullLayout._d3locale;
+        optsOut.text = [];
+        var pt;
+        if(Array.isArray(texttemplate)) {
+            for(i = 0; i < texttemplate.length; i++) {
```

Comment:
So here you're filling up to `texttemplate.length`. What does:

```js
Plotly.newPlot(gd, [{
  type: 'scattergl',
  y: [1, 2, 1],
  texttemplate: ['%{y} a', '%{y} b']
}])
```

show when hovering over the third pt?

Reply:
It errors out at the moment. Thanks for pointing this out...

Reply:
Although it seems like on `master`, it also gives out an error if the `text` is shorter than the data array:
![Screenshot_2019-08-13_14-10-16](https://user-images.githubusercontent.com/301546/62966297-be018d80-bdd4-11e9-8c69-d3325806826f.png)